### PR TITLE
OSDOCS 8020 modify the build_for_portal.py and makeBuild.py scripts to support second-level books and improve intra-book links

### PR DESCRIPTION
**Pull request by mramendi**
Version(s):

4.14+ to validate

Issue:

Link to docs preview:

QE review:
- No QE is required for this change.

Additional information:

### CODE REVIEW

File: build_for_portal.py [hunk 1]
```diff
@@ -269,19 +271,44 @@ def ensure_directory(directory):
     Creates DIRECTORY if it does not exist.
     """
     if not os.path.exists(directory):
-        os.mkdir(directory)
+        os.makedirs(directory)
+
+def expand_huge_books(info):
+    """
```
Comment (maxwelldb) [inline build_for_portal.py:277]: Would something like this work if you wanted to go mad with list comprehensions?

```
def expand_huge_books(info):
"""
Finds nodes for huge books, creates new nodes for books from their top-level topics,
and then removes the nodes for huge books
"""
huge_book_nodes = [book for book in info["book_nodes"] if book["Name"] in LIST_OF_HUGE_BOOKS]
additional_nodes = [
    {
        "Dir": f"{book['Dir']}/{topic['Dir']}",
        **topic,
    }
    for book in huge_book_nodes
    for topic in book["Topics"]
    if "Dir" in topic
]
info["huge_book_dirs"].extend(book["Dir"] for book in huge_book_nodes)
info["book_nodes"] = [node for node in info["book_nodes"] if node not in huge_book_nodes]
info["book_nodes"].extend(additional_nodes)
```

Could also consider pathlib for all of these changes if you like its ergonomics. 🤷

Comment (mramendi) [reply build_for_portal.py:274]: I'm not sure about going just *this* mad with list comprehensions but I will try to use some of this tomorrow Thursday. As for pathlib, I think it should be used everywhere or nowhere and it's to much work to use it everywhere.

**Comment by kalexand-rh** [comment]
@maxwelldb, do you see any reason to wait to merge this one?

**Comment by maxwelldb** [comment]
@kalexand-rh  Not really, no. 

It'd be nice to run Black on any Python code as a matter of neatness, but you're the DPM and can make a call as to whether you want to be strict about formatting or not. 👍

**Comment by kalexand-rh** [comment]
Then for the sake of orderliness, will one you do that?

**Comment by mramendi** [comment]
@maxwelldb while I don't have a problem with running Black, I suggest this be done separately, just in case Black alters the execution even though it is not designed to do so.

There is one more issue to work out anyway: the imp module, which is deprecated and slated to be removed. I can look into this - but again I'd like to separate this from the function enhancement.

**Comment by kalexand-rh** [comment]
Fair enough. I'm going to merge it and CP it to 4.14.

**Comment by kalexand-rh** [comment]
@mramendi, will you please manually cherry-pick to 4.14? I reviewed the commit history between `main` and `enterprise-4.14`. Usually CPs from `main` to the branch for the version that's about to be released need to go cleanly, but I think this is one instance where it will make more sense to manually CP.

**Comment by mramendi** [comment]
@kalexand-rh done #65981

**Comment by aireilly** [comment]
This change passes broken AsciiDoc and reports a passed build. IMO it should be reverted and reworked, tested thoroughly before an merge review again.

**Comment by mramendi** [comment]
@aireilly details please?

**Comment by kalexand-rh** [comment]
The sync to the portal uses build_for_portal.py, not build.py, and, at least as of Thursday, all of the 4.14 PV1 jobs were passing. I think we need to [investigate](https://issues.redhat.com/browse/OSDOCS-8050) switching Travis to use build_for_portal.py.

**Pull request merged**